shuup/core/api/service.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2017, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+
+from parler_rest.serializers import (
+    TranslatableModelSerializer, TranslatedFieldsField
+)
+from rest_framework.serializers import SerializerMethodField
+
+from shuup.core.models import PaymentMethod, ShippingMethod
+
+
+class ServiceSerializerMixin(object):
+    def get_price(self, service):
+        price = 0
+        request = self.context["request"]
+        if request.basket:
```

Comment:
You are assuming that we have the basket on each request, which will be false when not using the Front middleware.
IMO, you should read the `source` from the context and whoever creates the instance of this serializer, needs to pass the `source` to the context to receive a valid price.

```python

source = self.context.get("source")
if source:
    price = service.get_total_cost(source).taxful_price.value
```

then when creating the serializer on a ViewSet
```python
class SomeViewSet(...):
    def get_serializer_context(self):
        context = super(SomeViewSet, self).get_serializer_context()
        context['source'] = self.request.basket
        return context
```

or even creating the serializer manually:
```python
PaymentMethodSerializer(..., context={..., 'source': order})
```

After doing this, you'll need to change the viewsets which uses these serializers